src/modules/commander/commander.cpp

```diff
@@ -2055,20 +2063,17 @@ int commander_thread_main(int argc, char *argv[])
 				}
 			}
 
-
-			was_landed = land_detector.landed;
 			was_falling = land_detector.freefall;
 		}
 
-		// Check for auto-disarm
-		if (armed.armed && land_detector.landed && disarm_when_landed > 0) {
-			auto_disarm_hysteresis.set_state_and_update(true);
-		} else {
-			auto_disarm_hysteresis.set_state_and_update(false);
+		if (auto_disarm_enable && land_detector.landed) {
```

Comment:
You can keep `armed.armed && land_detector.landed && disarm_when_landed > 0)` here.